wasp/benchspy/metrics.go

```diff
@@ -0,0 +1,49 @@
+package benchspy
+
+import (
+	"math"
+	"sort"
+	"strconv"
+)
+
+// CalculatePercentile computes the specified percentile of a slice of numbers.
+// It is useful for statistical analysis, allowing users to understand data distributions
+// by retrieving values at specific percentiles, such as median or 95th percentile.
+func CalculatePercentile(numbers []float64, percentile float64) float64 {
```

Comment:
I'd better use this [lib](https://pkg.go.dev/github.com/montanaflynn/stats#section-readme) but up to you. If we'll need something in the future we better stick with libs for statistics

Reply:
gut idea, lets do it